### DISCUSSION TITLE
Chromium update with hardware video decode

### DISCUSSION THREAD

**Discussion by perfectsyrup** [Joshua-Riek/ubuntu-rockchip#271] [Q&A]
[Joshua Riek](https://github.com/Joshua-Riek), thank you for such a good Ubuntu experience, using it at NanoPC-T6 and everything is very smooth.

Only problem I have is to update installed chromium, becase you use 110.0.5481.4 and it has critical security bugs, so it is not recommended to use it in daily work or online banking.

Tried to intall latest chromium v.115 via flatpak and gnome software - all is ok, but no hardeware video decode is used even if I enable all GPU features in cromium and they are the same in chrome://gpu/ as at your build.

Do you use some custom build of chromium in your images?

Please help to update it and keep 4k60 hardware video decode feature.

**Comment by Joshua-Riek** [comment]
Chromium requires a hack to support the GPU on this platform. So yes, I use a custom build of Chromium in my images. 

Another developer has done work to patch and package Chromium. I'm not sure if you are a programmer but it will take some work to update. I have not looked into exactly how much effort this will be, but it will not be simple. 

Source: https://launchpad.net/~liujianfeng1994/+archive/ubuntu/rockchip-multimedia/+packages

**Comment by Joshua-Riek** [comment]
Chromium requires a hack to support the GPU on this platform. So yes, I use a custom build of Chromium in my images. 

Another developer has done work to patch and package Chromium. I'm not sure if you are a programmer but it will take some work to update. I have not looked into exactly how much effort this will be, but it will not be simple. 

Source: https://launchpad.net/~liujianfeng1994/+archive/ubuntu/rockchip-multimedia/+packages

**Reply by perfectsyrup** [reply]
I am not a programmer, so I contacted author of the custom build Chromium. V.115 of Chromium changed codebase a lot, so it is not an easy task to port his multimedia hacks to it - probably he will do it some time.

There are chromium hacks source code from Rockchip itself if it is any help for somebody interested in topic: 
https://github.com/JeffyCN/meta-rockchip/tree/master/dynamic-layers/recipes-browser/chromium

**Reply by avi951** [reply]
@perfectsyrup,
I have tried to build Chrome and apply the above patches, but there are still some hacks that I need to implement. but still not getting hardware acceleration in `chrome://gpu`

**Reply by Lolmc0587** [reply]
@avi951 you can build your chromium from here
https://github.com/amazingfate/chromium-debian-build/tree/rockchip-rkmpp-130
this work smoothly on this image

**Reply by avi951** [reply]
@Lolmc0587, Thank you for guiding me. I have tried to build Chromium 134.0.6998.88 using Debian source and getting rust version conflict.

rustc: `rustc 1.78.0`
clang version : `clang++19`

```
/mnt/debian/chromium-134.0.6998.88/out/Release$ sudo "python3" "../../build/rust/rustc_wrapper.py" --rustc=../../../../../usr/bin/rustc --depfile=obj/third_party/rust/simd_adler32/v0_3/lib/libsimd_adler32_lib.rlib.d --rsp=obj/third_party/rust/simd_adler32/v0_3/lib/libsimd_adler32_lib.rlib.rsp -- -Clinker="clang++-19" --crate-name simd_adler32 ../../third_party/rust/chromium_crates_io/vendor/simd-adler32-0.3.7/src/lib.rs --crate-type rlib --cap-lints=allow --cfg=feature=\"const-generics\" --cfg=feature=\"std\" -Cmetadata=simd_adler32-0.3 --edition=2018 -Cforce-unwind-tables=no -Crelocation-model=pic -Zsplit-lto-unit -Clinker-plugin-lto=yes -Cdefault-linker-libraries -Zdep-info-omit-d-target -Zmacro-backtrace -Zremap-cwd-prefix=. -Zexternal-clangrt --color=always --target=aarch64-unknown-linux-gnu -Ccodegen-units=1 --cfg cr_rustc_revision=\"rustc\ 1.78.0\" -Copt-level=3 -Zallow-features= -Clinker-plugin-lto=yes -Dunsafe_op_in_unsafe_fn --emit=dep-info=obj/third_party/rust/simd_adler32/v0_3/lib/libsimd_adler32_lib.rlib.d,link -o obj/third_party/rust/simd_adler32/v0_3/lib/libsimd_adler32_lib.rlib LDFLAGS RUSTENV OUT_DIR=../../../../../../out/Release/gen/third_party/rust/simd_adler32/v0_3/lib CARGO_PKG_AUTHORS=Marvin\ Countryman\ \<me@maar.vin\> CARGO_PKG_VERSION=0.3.7 CARGO_PKG_NAME=simd-adler32 CARGO_PKG_DESCRIPTION=A\ SIMD-accelerated\ Adler-32\ hash\ algorithm\ implementation. CARGO_MANIFEST_DIR=../../third_party/rust/simd_adler32/v0_3/lib/crate
error: the option `Z` is only accept/mnt/debian/chromium-134.0.6998.88/out/Release$ sudo "python3" "../../build/rust/rustc_wrapper.py" --rustc=../../../../../usr/bin/rustc --depfile=obj/third_party/rust/simd_adler32/v0_3/lib/libsimd_adler32_lib.rlib.d --rsp=obj/third_party/rust/simd_adler32/v0_3/lib/libsimd_adler32_lib.rlib.rsp -- -Clinker="clang++-19" --crate-name simd_adler32 ../../third_party/rust/chromium_crates_io/vendor/simd-adler32-0.3.7/src/lib.rs --crate-type rlib --cap-lints=allow --cfg=feature=\"const-generics\" --cfg=feature=\"std\" -Cmetadata=simd_adler32-0.3 --edition=2018 -Cforce-unwind-tables=no -Crelocation-model=pic -Zsplit-lto-unit -Clinker-plugin-lto=yes -Cdefault-linker-libraries -Zdep-info-omit-d-target -Zmacro-backtrace -Zremap-cwd-prefix=. -Zexternal-clangrt --color=always --target=aarch64-unknown-linux-gnu -Ccodegen-units=1 --cfg cr_rustc_revision=\"rustc\ 1.78.0\" -Copt-level=3 -Zallow-features= -Clinker-plugin-lto=yes -Dunsafe_op_in_unsafe_fn --emit=dep-info=obj/third_party/rust/simd_adler32/v0_3/lib/libsimd_adler32_lib.rlib.d,link -o obj/third_party/rust/simd_adler32/v0_3/lib/libsimd_adler32_lib.rlib LDFLAGS RUSTENV OUT_DIR=../../../../../../out/Release/gen/third_party/rust/simd_adler32/v0_3/lib CARGO_PKG_AUTHORS=Marvin\ Countryman\ \<me@maar.vin\> CARGO_PKG_VERSION=0.3.7 CARGO_PKG_NAME=simd-adler32 CARGO_PKG_DESCRIPTION=A\ SIMD-accelerated\ Adler-32\ hash\ algorithm\ implementation. CARGO_MANIFEST_DIR=../../third_party/rust/simd_adler32/v0_3/lib/crate
error: the option `Z` is only accepted on the nightly compiler

help: consider switching to a nightly toolchain: `rustup default nightly`

note: selecting a toolchain with `+toolchain` arguments require a rustup proxy; see <https://rust-lang.github.io/rustup/concepts/index.html>

note: for more information about Rust's stability policy, see <https://doc.rust-lang.org/book/appendix-07-nightly-rust.html#unstable-features>

error: 1 nightly option were parseded on the nightly compiler

help: consider switching to a nightly toolchain: `rustup default nightly`

note: selecting a toolchain with `+toolchain` arguments require a rustup proxy; see <https://rust-lang.github.io/rustup/concepts/index.html>

note: for more information about Rust's stability policy, see <https://doc.rust-lang.org/book/appendix-07-nightly-rust.html#unstable-features>

error: 1 nightly option were parsed
```

**Reply by Lolmc0587** [reply]
No, the version https://github.com/amazingfate/chromium-debian-build/tree/rockchip-rkmpp-130 working well on this image. Any version above does not working properly
If you can't install, i can give you prebuilt chromium 130
https://www.mediafire.com/file/bupa2qvqpjotu40/chromium_130_v42l.zip/file
